text/0000-orderer-pruning-and-snapshot.md

```diff
@@ -0,0 +1,275 @@
+---
+layout: default
+title: Orderer ledger pruning/snapshots
+nav_order: 3
+---
+
+- Feature Name: OSN_pruning_snapshots
+- Start Date: 2021-03-23
+- RFC PR:
+- Fabric Component: orderer, ledger
+- Fabric Issue:
+
+# Summary
+[summary]: #summary
+
+The RFC introduces ledger snapshot feature at orderers, similar to the same feature at peers. Additionally the RFC describes the proposed process of pruning the ledger without restarting the orderer.
+
+# Motivation
+[motivation]: #motivation
+
+Real world use cases show the necessity of controlling the disk space consumed by block ledger, and this RFC is mainly motivated by the need to reduce space consumption at orderer nodes.
+Additionally this RFC presents a better way to join a channel without pulling the huge number of all historical blocks, which also may take significant time and may pose a problem for scaling of an existing Fabric network.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This RFC introduces a notion of a **ledger snapshot for OSNs** (ordering service nodes). With this notion three new logical operations appear at the orderer REST API and in the CLI tool *osnadmin*:
+- Create a snapshot,
+- Join channel from a snapshot,
+- Prune block ledger of an existing channel.
+
+Ledger snapshot is a set of files containing all the necessary information for OSN to join a channel from a block later than a genesis block (block #0) without loss of any functionality except the ability to deliver an earlier block.
+OSN ledger snapshot functionality is designed with the same principles as the peer ledger snapshot, in particular it is a full snapshot. We do not implement an incremental snapshot capability.
```

Comment:
> as the peer ledger snapshot, in particular it is a full snapshot. We do not implement an incremental snapshot capability.

I am not sure what do you mean by "incremental snapshot" in the context of an OSN. (In the case of a peer, it refers mainly to the contents of the statedb)

Reply:
> I am not sure what do you mean by "incremental snapshot" in the context of an OSN
I mean "any built-in mechanism that allows referring to an info from a previous snapshot instead of explicitly including it into the current snapshot"

Mostly this means that we do not include any mechanism of the possible dependency on a previous snapshot in the snapshot format.
I agree that there is no real reason for having incremental snapshots in OSNs, at least at the moment. So we just mention the absence of any mechanism built into the snapshot format for incremental build-up because we want to highlight the conceptual similarity between peer snapshots and OSN snapshots.

Anyway, I have improved the text to make it absolutely clear in this matter. Please check the latest commit.

Also it is very minor issue, so if you think it creates unnecessary confusion, I am happy to remove this paragraph altogether.

Reply:
Yes, I feel that this adds an unnecessary diversion. So, better to omit it altogether in my opinion.

Reply:
Yes, I feel that it causes unnecessary diversion and can be omitted entirely.

Reply:
Done, removed here:
b8f85b8ffd241e64a22b8e3dbc9be61aff9f8714